### PR TITLE
fix(electrostatics): retain Ewald Miller topology

### LINKED ISSUE #147: 🚀[FEA]: Address incorrect virial in `ewald_reciprocal_space()` with detached cells
### Is this a new feature, an improvement, or a change to existing functionality?

Improvement

### How would you describe the priority of this feature request

Low (would be nice)

### Please provide a clear description of problem you would like to solve.

As pointed out in issue #136, taking a strain/cell gradient through `ewald_reciprocal_space()` produces an incorrect virial when using precomputed `k_vectors` (i.e. detached cell). In nvalchemiops 0.4.1 a `UserWarning` is now raised (see PR #138). 

An alternative solution to prevent this would be to modify `generate_k_vectors_ewald_summation()` to return `miller_indices` instead of `k_vectors` (since this is what is computed internally). Then `ewald_reciprocal_space()` could compute `k_vectors` internally as:
```python
reciprocal_cell = (
    TWOPI * torch.linalg.inv_ex(cell.transpose(1, 2))[0]
)  # Transpose for column vectors
k_vectors = miller_indices @ reciprocal_cell
```

### Describe any alternatives you have considered

This can be handled manually with the following workaround:
```python
# precompute miller indices
k_vectors = generate_k_vectors_ewald_summation(cell=cell, k_cutoff=k_cutoff)
k_miller = torch.round(
    torch.bmm(k_vectors, cell.detach().transpose(-1, -2)) / TWOPI 
)
...
# rebuild from the cell so ∂k/∂cell exists to get correct virial
k_vectors = TWOPI * torch.matmul(
    k_miller.to(cell.dtype),
    torch.linalg.inv(cell).transpose(-1, -2),
)
```

### PR DESCRIPTION
**Pull request by zubatyuk**
# ALCHEMI Toolkit-Ops Pull Request

## Description

Retain Ewald Miller topology separately from Cartesian reciprocal vectors, so changing-cell calls materialize vectors from the current cell. Fix JAX reciprocal-component autodiff to preserve the tangent carried by graph-derived `k_vectors`.

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Performance improvement
- [x] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] CI/CD or infrastructure change

## Related Issues

Fixes #147.

## Changes Made

- Add caller-retained Miller-index generators, live-cell materializers, and retained-index reciprocal components for Torch and JAX.
- Extend full Ewald with `miller_indices=` while preserving explicit-vector metadata behavior.
- Correct JAX reciprocal-component derivatives for graph-derived vectors and document fixed-vector semantics.
- Cover public topology, derivative, batch/JIT, direct-output, and compatibility behavior.

## Testing

- [x] Unit tests pass locally (`make pytest`)
- [x] Linting passes (`make lint`)
- [x] New tests added for new functionality meets coverage expectations?

## Checklist

- [x] I have read and understand the [Contributing Guidelines](../CONTRIBUTING.md)
- [x] I have updated the [CHANGELOG.md](../CHANGELOG.md)
- [x] I have performed a self-review of my code
- [x] I have added docstrings to new functions/classes
- [x] I have updated the documentation (if applicable)

## Additional Notes

None.

> [!TIP]
> This repository uses Greptile, an AI code review service, to help conduct
> pull request reviews. We encourage contributors to read and consider suggestions
> made by Greptile, but note that human maintainers will provide the necessary
> reviews for merging: Greptile's comments are **not** a qualitative judgement
> of your code, nor is it an indication that the PR will be accepted/rejected.
> We encourage the use of emoji reactions to Greptile comments, depending on
> their usefulness and accuracy.

### CODE REVIEW

**Review by dallasfoster — APPROVED** [review]

**Comment by dallasfoster** [comment]
/ok to test [c33a04b](https://github.com/NVIDIA/nvalchemi-toolkit-ops/pull/152/commits/c33a04bd8e7a6e47baac773809d80356167ba44a)

**Comment by dallasfoster** [comment]
/ok to test [48532cf](https://github.com/NVIDIA/nvalchemi-toolkit-ops/pull/152/commits/48532cfcc5948a32dc294920e3b18c8370708867)

**Review by dallasfoster — APPROVED** [review]
